### ISSUE TITLE
False-positive returning address of static local

### ISSUE THREAD

**Issue by mgehre** [mgehre/llvm-project#20]
```
const char *
f()
{
    static char path[256];
   return path;
}
```
gives `warning: returning a dangling Pointer [-Wlifetime]` with `-Wdangling-lifetime-pointer -Waddress-of-temporary`
